packages/gestalt/src/ComboBox.js

```diff
@@ -11,59 +11,136 @@ import {
+   */
   errorMessage?: Node,
+  /**
+   * Provides additional information about how to select a ComboBox option.
+   */
   helperText?: string,
+  /**
+   * The user input in ComboBox for controlled components. See [controlled ComboBox](https://gestalt.pinterest.systems/combobox#Controlled-vs-Uncontrolled) variant to learn more.
+   */
   inputValue?: string | null,
+  /**
+   * Unique id to identify each ComboBox. Used for [accessibility](https://gestalt.pinterest.systems/combobox#Accessibility) purposes.
+   */
+  id: string,
+  /**
+   * Provide a label to identify the ComboBox field.
+   */
+  label: string,
+  /**
+   * Whether the label should be visible or not. If `hidden`, the label is still available for screen reader users, but does not appear visually. See the [label visibility variant](https://gestalt.pinterest.systems/combobox#Label-visibility) for more info.
+   */
   labelDisplay?: 'visible' | 'hidden',
+  /**
+   * The text shown when the input value returns no matches.
+   */
+  noResultText: string,
+  /**
+   * Callback when you focus outside the component.
+   */
   onBlur?: ({|
     event: SyntheticFocusEvent<HTMLInputElement> | SyntheticEvent<HTMLInputElement>,
     value: string,
   |}) => void,
+  /**
+   * Callback when user types into the control input field.
+   */
   onChange?: ({|
-    value: string,
     event: SyntheticInputEvent<HTMLInputElement>,
+    value: string,
   |}) => void,
+  /**
+   * Callback when user clicks on clear button.
+   */
   onClear?: () => void,
+  /**
+   * Callback when you focus on the component.
+   */
   onFocus?: ({|
     event: SyntheticFocusEvent<HTMLInputElement>,
     value: string,
   |}) => void,
+  /**
+   * Callback for key stroke events. See [tags](#Tags) variant to learn more.
+   */
   onKeyDown?: ({|
     event: SyntheticKeyboardEvent<HTMLInputElement>,
     value: string,
   |}) => void,
+  /**
+   * Callback when an item is selected.
+   */
   onSelect?: ({|
     event: SyntheticInputEvent<HTMLElement> | SyntheticKeyboardEvent<HTMLElement>,
-    item: ComboBoxItemType,
+    item: {|
+      label: string,
+      subtext?: string,
+      value: string,
+    |},
   |}) => void,
+  /**
+   * The data for each selection option. See [subtext](https://gestalt.pinterest.systems/combobox#With-subtext) variant to learn more.
+   */
+  options: $ReadOnlyArray<{|
+    label: string,
+    subtext?: string,
+    value: string,
+  |}>,
+  /**
+   * Specify a short description that suggests the expected input for the field.
+   */
   placeholder?: string,
-  selectedOption?: ComboBoxItemType,
+  // The ref prop is unused and listed here just for documentation purposes.
+  /**
+   * Forward the ref to the underlying component container element. See the [Ref](https://gestalt.pinterest.systems/combobox#Ref) variant to learn more about focus management.
+   */
+  ref?: Ref<'input'>, // eslint-disable-line react/no-unused-prop-types
```

Comment:
I hadn't noted this previously as I'd commented about it on another PR, but can you add logic to display this as `React.Ref<'input'>`?

Reply:
DONE